src/Microsoft.ML.Transforms/Text/TextFeaturizingEstimator.cs

```diff
@@ -416,13 +416,14 @@ public ITransformer Fit(IDataView input)
             string charFeatureCol = null;
             List<string> tempCols = new List<string>();
             IDataView view = input;
+            TransformerChain<ITransformer> chain = new TransformerChain<ITransformer>();
 
```

Comment:
Why not building an `EstimatorChain` along the way according to the specified options instead of a `TransformerChain`?
You are essentially creating a pipeline of other estimators so it could make sense to act at that level. If you create a pipeline of estimators and call `.Fit(input)` at the end, the `EstimatorChain.Fit()` method runs code that is essentially the same as what you do in `AddToChainAndTransform`. 

Did you choose this approach because some of the transforms don't have an estimator version?

Reply:
I tried to follow your suggestion and create an `EstimatorChain` instead of a `TransformerChain`, but I got stuck in the drop-missing-values part (WordBagTransform.cs line 326). It turns out that the `MissingValueDroppingTransformer` doesn't have an estimator.

---
In reply to: [272266922](https://github.com/dotnet/machinelearning/pull/3185#discussion_r272266922) [](ancestors = 272266922)